libutils/string_lib.c

```diff
@@ -60,6 +60,72 @@ char *StringFormat(const char *fmt, ...)
     return res;
 }
 
+static char *stpncpy_fallback(char *dst, const char *src, size_t len)
+{
+    assert(dst != NULL);
+    assert(src != NULL);
+    assert(dst != src);
+
+    for (int i = 0; i < len; ++i)
+    {
+        const char copy_byte = src[i];
+        dst[i] = copy_byte;
+        if (copy_byte == '\0')
+        {
+            // Zero fill and return:
+            for (int j = i+1; j < len; ++j)
+            {
+                dst[j] = '\0';
+            }
+            return dst + i;
+        }
+    }
+    return dst + len;
+}
```

Comment:
Why is this needed here? It is the same as the implementation in *libcompat*, right?

Reply:
Yes, thanks, removed. I just added it while debugging.